hathor/metrics.py

```diff
@@ -322,10 +337,33 @@ def set_cache_data(self) -> None:
             if misses:
                 self.transaction_cache_misses = misses
 
+    def set_tx_storage_data(self) -> None:
+        store = self.tx_storage
+
+        if isinstance(self.tx_storage, TransactionCacheStorage):
+            store = self.tx_storage.store
+
+        if not isinstance(store, TransactionRocksDBStorage):
+            # We currently only collect metrics for RocksDB
+            return
+
+        # Try to use the number of blocks to decide when we should update the tx storage data
+        block_count = self.tx_storage.get_height_best_block()
```

Comment:
Note that `block_count` might be different from `best_block.height` since we might have orphan blocks. I guess you meant `best_block_count`, right?

Reply:
Yes, `best_block_count` would describe it more accurately

Reply:
Fixed

Reply:
Thanks!